apis/v1alpha1/httproute_types.go

```diff
@@ -27,16 +27,24 @@ import (
 
 // HTTPRoute is the Schema for the HTTPRoute resource.
 type HTTPRoute struct {
-	metav1.TypeMeta   `json:",inline"`
+	metav1.TypeMeta `json:",inline"`
+	// +optional
 	metav1.ObjectMeta `json:"metadata,omitempty"`
 
-	Spec   HTTPRouteSpec   `json:"spec,omitempty"`
+	// Spec defines the desired state of HTTPRoute.
+	//
+	// +optional
+	Spec HTTPRouteSpec `json:"spec,omitempty"`
```

Comment:
Same comment as before. We should remove `omitempty` tag from Spec everywhere rather than adding an `+optional`.

Reply:
Same response as https://github.com/kubernetes-sigs/service-apis/pull/538/files#r573947867.